### DISCUSSION TITLE
URL created application has a Project path BUG

### DISCUSSION THREAD

**Discussion by omar-chaabouni** [codemagic-ci-cd/codemagic-docs#1913] [Q&A] [answered]
### Use case

```Text
I found a BUG, I thought I share it for better user experience.
When you add an applicaiton with git URL for example git@bitbucket.org:test1/testapp.git 
The project gets added and after adding the web hook, I try to change Project path under build section
but the project path wouldn't change, unless I launched the first build. 
Which obviously gave an error and then I was able to change the project path.
The bug could be persisent not only with URL added apps, I don't know.
```

### Idea

```Text
I hope Codemagic fix it, shouldn't be that hard.
```

**Comment by icarusdust** [comment]
Hi @omar-chaabouni, I cannot reproduce it. When I am adding an app though its URL, Change Settings option gets available, no need to trigger a build to see it.

**Reply by omar-chaabouni** [reply]
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/55398565/e89cba66-ba63-4a09-bf41-348085b959d9)

I can't change project path here, I should find a driectory here (in which I have my code)
as a first build this will always be a fail
once first build finished I can go back to change settings and then I will be able to find my project directory I am looking for
The problem is with project path, when you first connect your project, it just doesn't give you the path until you do the first build

**Reply by icarusdust** [reply]
Thanks for further elaborating on it. There is an easy way to scan your project paths by clicking the re-scan button without triggering any initial builds as shown in the image below. However, please note that it is available only with repos that are connected with OAuth and as I see your repo is connected through SSH manually. Try it with OAuth and see if it helps you
<img width="866" alt="Screenshot 2023-08-01 at 15 07 41" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/43378371/7f4dd5d6-95d7-4b29-82f8-448f4d07bb07">

**Reply by omar-chaabouni** [reply]
@icarusdust I remember doing that it didn't work. I don't know.

**Reply by icarusdust** [reply]
What do you exactly mean by it did not work ? You mean you connected through OAuth and the button did not appear ?

**Reply by omar-chaabouni** [reply]
what do you exacly mean by OAuth

**Reply by icarusdust** [reply]
Adding your repo through Bitbucket OAuth by connecting your Bitbucket account in the team settings of the UI instead of using its url through the Other option when adding apps.

**Reply by omar-chaabouni** [reply]
Yes, I did that first thing after creating the account. It was already done but the project was created via URL: git@....

**Reply by icarusdust** [reply]
@omar-chaabouni please share your team id where I can find that you connected BB through OAuth and there is an app added to your account through that OAuth, not manually via its SSH url.

**Reply by icarusdust** [reply]
and in general, back to our main thread topic, it is not a bug, it is how it is supposed to behave when connecting repos through SSH.

**Reply by icarusdust** [reply, marked as answer]
It is mentioned [in the docs](https://docs.codemagic.io/flutter-configuration/flutter-projects/#multiple-flutter-projects-in-one-repository) as well that rescanning is not available for repositories added from self-hosted and repos added via SSH. Instead, project paths are scanned on each new build and are updated on change.
